crates/common/fork_choice/src/handlers.rs

```diff
@@ -145,10 +141,24 @@ pub async fn on_block(
     let block = &signed_block.message;
 
     // Parent block must be known
-    ensure!(store.block_states.contains_key(&block.parent_root));
+    if !store.block_states.contains_key(&block.parent_root) {
+        println!(
+            "Missing parent block state for parent_root: {:x}",
+            block.parent_root
+        );
+        return Ok(());
+    }
```

Comment:
```suggestion
ensure!(store.block_states.contains_key(&block.parent_root), "Missing parent block state for parent_root: {:x}", block.parent_root);
```

This is supposed to throw an error